### PR TITLE
Breakdown trees3.0

### PR DESCRIPTION
**Pull request by MeghaCBorty**
Update to iTwin.js 3.0.
Implement and Fix Eslint.

### CODE REVIEW

File: packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/CombinedTreeNodeFunctionalityProvider.ts
Comment (juliusBaranauskas) [inline packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/CombinedTreeNodeFunctionalityProvider.ts:40]: here and in other places of this file a whitespace is missing `for (const i of x){`

File: packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx
Comment (juliusBaranauskas) [inline packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx:130]: `enableVisibility={!!props.enableVisibility}` would give the same result

File: packages/itwin/breakdown-trees/src/Views/TreeNodeFunctionsToolbar.tsx
Comment (juliusBaranauskas) [inline packages/itwin/breakdown-trees/src/Views/TreeNodeFunctionsToolbar.tsx:32]: for useMemo's that are not expected to return a value (just run asynchronously) a `void` operator can be added in front to indicate that it does not await the Promise on purpose, e.g.:
```
void React.useMemo(async () => {...}, [...]);
```
That way the linter will not complain

File: README.md
Comment (aruniverse) [inline README.md:18]: pkg src needs to move into the itwin directory

the badges should point to itwin not bentley

File: README.md
Comment (aruniverse) [inline README.md:26]: looks like you added an extra col (L23 too)

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:3]: reset the ver to 0.1.0

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:48]: should be able to remove itwin-client, product-settings-client

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:105]: need to review peer deps here

File: packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/SelectRelatedFunctionalityProvider.ts
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/SelectRelatedFunctionalityProvider.ts:17]: why is this potentially undefined?

**Review by aruniverse — COMMENTED** [review]
Ill do another pass today reviewing the deps;

I saw some files had no changes (besides the move from bentley to itwin dir). Please double check the scss files, as those will most likely need to be updated if they ever imported scss from core

**Comment by aruniverse** [comment]
@MeghaCBorty , should @juliusBaranauskas be a codeowner?

File: packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/SelectRelatedFunctionalityProvider.ts
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/SelectRelatedFunctionalityProvider.ts:17]: Yeah, this one can be any additional event handler user wants to execute, like from factorysight we use it for tracking in appInsights.. So it can be undefined if user does not need any additional functionality like this.  Previously when it was a mandate, we were handling the call with !, but with eslint its throwing error @typescript-eslint/no-non-null-asserted-optional-chain. So decided to allow undefined here.

File: packages/itwin/breakdown-trees/package.json
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/package.json]: done

**Comment by MeghaCBorty** [comment]
> @MeghaCBorty , should @juliusBaranauskas be a codeowner?

Not needed now.

File: rush.json
Comment (aruniverse) [inline rush.json:35]: i would rename the version policy too; think it might cause an issue when publishing

File: packages/itwin/breakdown-trees/tsconfig.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/tsconfig.json:2]: any reason for needing to add the types?

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:2]: url on L12 needs to be updated

Comment (aruniverse) [reply packages/itwin/breakdown-trees/package.json:2]: are you guys going to be delivering cjs & esm like all the other pkgs?

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:107]: can just be a dev dep

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:108]: remove

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:109]: remove

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:118]: remove

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:119]: whats the need for this pkg?

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:120]: i _think_ you can do `workspace:^*`

Comment (aruniverse) [reply packages/itwin/breakdown-trees/package.json:120]: could you try that out and lmk if that works?

File: packages/itwin/breakdown-trees/src/assets/ClassificationSystems.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/assets/ClassificationSystems.json:3]: could you update all these ruleset ids, so theyre prefixed with itwin-breakdown-trees instead?

File: packages/itwin/breakdown-trees/src/Views/NodeRenderers/FunctionalTreeNodeRenderer.tsx
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/NodeRenderers/FunctionalTreeNodeRenderer.tsx:41]: did this line need to change? a little redundant

File: packages/itwin/breakdown-trees/src/Views/visibility/SpaceLabelDecorator.ts
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/visibility/SpaceLabelDecorator.ts:92]: not the safest... but will leave it up to you guys to fix

**Review by aruniverse — COMMENTED** [review]
major thing you should update before this pr can go through are the peer deps

File: packages/itwin/breakdown-trees/tsconfig.json
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/tsconfig.json:2]: yes, we reuse some of the configurations from there and removing that will create huge number of build errors.

File: packages/itwin/breakdown-trees/src/Views/NodeRenderers/FunctionalTreeNodeRenderer.tsx
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/src/Views/NodeRenderers/FunctionalTreeNodeRenderer.tsx:41]: Yes, visibilityHandler can have value but we won't pass it here if enableVisibility is false.

File: packages/itwin/breakdown-trees/package.json
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/package.json:120]: Yes, this works and we had it before.. But in my last PR you suggested to use it this way in peer deps as the iModel repo does.. So I changed it to this.

File: rush.json
Comment (MeghaCBorty) [reply rush.json:35]: renamed

File: packages/itwin/breakdown-trees/tsconfig.json
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/tsconfig.json]: This was here for long time. Just removed it and had no issue.

File: packages/itwin/breakdown-trees/package.json
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/package.json]: Removed

File: packages/itwin/breakdown-trees/package.json
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/package.json:2]: Thanks for raising the concern. Just fixed our build scripts and pushed a commit to generate cjs and esm.

File: packages/itwin/breakdown-trees/package.json
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:30]: i think this line needs to get updated to call copy cjs & esm

**Review by aruniverse — APPROVED** [review]
I think just need to fix the copy scss assets script, but other than that should be good to go

File: packages/itwin/breakdown-trees/package.json
Comment (MeghaCBorty) [reply packages/itwin/breakdown-trees/package.json:30]: copy cjs and esm are called from the build script and removing that from copy:assets is causing test failure.So keeping it there.

**Pull request merged**
